### ISSUE TITLE
Handling Molecular Event - discussion

### ISSUE THREAD

**Issue by goodb** [geneontology/go-shapes#232]
As a result of https://github.com/geneontology/pathways2GO/issues/98 , I have introduced a new class into the REACTO ontology called 'Molecular Event'.  It is a subclass of BFO Process and could, in principle, be a superclass of GO Molecular Function.  This is used to describe biological activities that may or may not have enabling gene product entities.  If they do have enablers, then these would be reclassified as GO Molecular Functions.  If not, then a new class of spontaneous processes may be needed.  This class may wind its way into the GO.  

We need to decide if we want to make any changes to the shex schema based on its addition.  Fundamentally, do we treat molecular events differently to molecular functions or not and how.  If we do nothing, the only immediate consequence would be that many models imported from Reactome would fail schema validation.  That could be a desired outcome, but if it is not, we need to change the schema.  (e.g. constraints on the MF shape that 'provides input for' or 'regulates' can only link to another MF would break when the downstream node is typed as a ME).  

@cmungall @thomaspd @ukemi @vanaukenk I'd love to have a quick answer here so I can run the validator on the reactome models and finish that paper.

**Comment by goodb** [comment]
After discussion with @cmungall the decision, for now, is to allow the use of Molecular Events in all the places where Molecular Function is now specified.  This will be handled in the schema using union statements.  In the future, the Molecular Event class may be used with the GO as part of an upper level ontology in between GO and BFO.  When that happens, we can update the shex schema to take advantage of the structure there (e.g. pointing to ME and its children, which will contain MF).

**Comment by ukemi** [comment]
Hi @goodb 
Do you mean Molecular Function is **not** specified?

**Comment by deustp01** [comment]
> a new class of spontaneous processes may be needed

If the Reactome sample of events is representative, truly spontaneous activities / reactions are very rare (but there are examples - [1PYR-5COOH spontaneously hydrolyses to L-GluSS](https://reactome.org/PathwayBrowser/#/R-HSA-70688&SEL=R-HSA-70667) - so this class will hold permanent instances as well as a much larger number that lack enablers now because of lacking experimental data (chunks of eicosanoid metabolism) or because we don't yet know how to identify the enabler in a way that GO can accommodate (dissociation events).

**Comment by deustp01** [comment]
> the only immediate consequence would be that many models imported from Reactome would fail schema validation. That could be a desired outcome, but if it is not, we need to change the schema. (e.g. constraints on the MF shape that 'provides input for' or 'regulates' can only link to another MF would break when the downstream node is typed as a ME).

This directly relates to the discussion in the pathways2GO manuscript of the usability of incomplete GO-CAM models. From Ben's description, it seems like information is lost if a complete activity unit can't be functionally linked to an incomplete downstream one. If this understanding is correct then it looks like it could be an argument for changing the schema? (But if these "provides-input" and "regulates" links can be preserved even if the model fails validation, then maybe not. I'm out of my depth here.)

**Comment by goodb** [comment]
@deustp01 yes, information is lost if we take any nodes out of the network.  We've discussed inferring connections across the gaps that would be created by taking those nodes out, but this is a bridge I see no need to cross.  If those nodes had no meaning or use, they wouldn't be in Reactome.

@ukemi no, Molecular Function is still a specified shape and still used in the same way for the shape definitions.  The proposed (see #232 ) change is to add in the molecular event shape (currently with no constraints apart from being an instance of the ME class) and, where MF is used in a constraint, also allow ME.

**Comment by goodb** [comment]
de ja vu from March 2020 - https://github.com/geneontology/go-shapes/issues/214 - assuming this change is accepted, close that issue too.
